elasticsearch-base/src/main/java/org/vertexium/elasticsearch/ElasticSearchSearchIndexConfiguration.java

```diff
@@ -157,4 +171,15 @@ private static ScoringStrategy getScoringStrategy(GraphConfiguration config) {
         String className = config.getString(GraphConfiguration.SEARCH_INDEX_PROP_PREFIX + "." + CONFIG_SCORING_STRATEGY_CLASS_NAME, DEFAULT_SCORING_STRATEGY.getName());
         return ConfigurationUtils.createProvider(className, config);
     }
+
+    private static NameSubstitutionStrategy getNameSubstitutionStrategy(GraphConfiguration config) {
+        String className = config.getString(GraphConfiguration.SEARCH_INDEX_PROP_PREFIX + "." + CONFIG_NAME_SUBSTITUTION_STRATEGY_CLASS_NAME, DEFAULT_NAME_SUBSTITUTION_STRATEGY.getName());
+        NameSubstitutionStrategy strategy = ConfigurationUtils.createProvider(className, config);
+
+        if (strategy instanceof SimpleNameSubstitutionStrategy) {
```

Comment:
Should we have a method on NameSubstitutionStrategy called config or setUp which takes a configuration map.